src/py_pecg/filtering_module.py

```diff
@@ -0,0 +1,564 @@
+        messages["setup"]["wavedet"]["freq"], messages
+    )
+    lengths = [len(q) for q in quadratic_filter_array]
+    decimation_values = [int(np.floor((l - 1) / 2)) for l in lengths]
+    quadratic_filter_dictionary = {
+        "filters": quadratic_filter_array,
+        "lengths": lengths,
+        "d_values": decimation_values,
+    }
+    return quadratic_filter_dictionary
+
+
+def filter_segment_of_signal(
+    quadratic_filter_dictionary: dict,
+    current_samp: int,
+    initial_samp: int,
+    num_samps: int,
+    segment_boundaries: list,
+    signal_segment: np.ndarray,
+    messages: dict,
+):
+    """
+    Uses a Wavelet Transform on individual ECG signal excerpts, after
+    Generating quadratic-spline filter banks.
+
+    Args:
+        quadratic_filter_dictionary (dict): A dictionary relevant to the algorithm filter-bank.
+        current_samp (int): Current sample analyzed within the ECG signal.
+        initial_samp (int): Initial sample analyzed within the ECG signal.
+        num_samps (int): Number of samples per excerpt for the filter bank corresponding to 2^16 samples at sf=250.
+        segment_boundaries (list): First and last indices analyzed within the ECG signal.
+        signal_segment (np.ndarray): The ECG signal designated for analysis.
+        messages (dict): Dictionary containing information relevant to the algorithm.
+
+    Returns:
+        signal_segment (np.ndarray): The ECG signal designated for analysis.
+        current_samp (int): Updated sample analyzed within the ECG signal.
+        updated_samp (int): Updated current sample analyzed within the ECG signal.
+        synchronized_wavelet_matrix (np.ndarray): Wavelet Transform Matrix, relevant to the current segment of the signal.
+        threshold_matrix (np.ndarray): QRS detection threshold matrix.
+        end_samp (int): Updated final sample analyzed within the ECG signal.
+        initial_samp (int): Updated initial sample analyzed within the ECG signal.
+
+    NOTE:
+        First l5-1 samples are not correctly filtered (border effect)
+        Last d5 samples are discarded in order to alineate all the
+        filtered signals, taking into acount the filter delays.
+    """
+
+    def process_filters(quadratic_filter_dictionary):
+        required_keys = {"filters", "lengths", "d_values"}
+        keys_in_dict = set(quadratic_filter_dictionary.keys())
+        missing_keys = required_keys - keys_in_dict
+        extra_keys = keys_in_dict - required_keys
+        if missing_keys:
+            raise ValueError(
+                f"quadratic_filter_dictionary is missing keys: {missing_keys}"
+            )
+        if extra_keys:
+            raise ValueError(
+                f"quadratic_filter_dictionary contains unexpected keys: {extra_keys}"
+            )
+        quadratic_filter_array, lengths, decimation_values = [
+            quadratic_filter_dictionary[key] for key in required_keys
+        ]
```

Comment:
we could just require `quadratic_filter_array, lengths, decimation_values` as separate variables and avoid this checks